packages/babel-plugin-proposal-pipeline-operator/test/fixtures/hack-caret/pipe-body-with-arrow-function-and-nested-pipe/output.js

```diff
@@ -1,8 +1,8 @@
-var _ref4, _ref5, _ref6;
+var _ref2;
 
-const result = (_ref6 = 5, (_ref5 = Math.pow(_ref6, 2), (_ref4 = [1, 2, 3].map(n => {
-  var _ref, _ref2, _ref3;
+const result = (_ref2 = Math.pow(5, 2), [1, 2, 3].map(n => {
+  var _ref;
 
-  return _ref3 = n + _ref5, (_ref2 = _ref3 * 2, (_ref = `${_ref2} apples`, _ref.toUpperCase()));
-}), _ref4.join())));
+  return _ref = (n + _ref2) * 2, `${_ref} apples`.toUpperCase();
```

Comment:
Template literal string parts are not marked as pure in `@babel/types`, I'll open a follow-up PR so that this can be compiled to
```js
return `${(n + _ref2) * 2} apples`.toUpperCase();
```